projs/test-game/src/GameModule.cpp

```diff
@@ -23,14 +26,28 @@ std::unique_ptr<vlkx::PushConstant> trans_constant_;
 std::unique_ptr<vlkxtemp::Model> cube_model_;
 float aspectRatio;
 
-void GameModule::PreInit() {  }
+std::shared_ptr<ShadowEngine::RendererModule> renderer;
+
+void GameModule::PreInit() {
+    spdlog::info("{0} PreInit", this->GetName());
+}
 
 void GameModule::Init() {
+    spdlog::info("{0} Init", this->GetName());
     spdlog::info("Game Module loading level..");
+
     trans_constant_ = std::make_unique<vlkx::PushConstant>(
             sizeof(Transformation), 2);
 
-    auto extent = ShadowEngine::ModuleManager::getInstance()->renderer->GetRenderExtent();
+    auto rendererPtr = ShadowEngine::ShadowApplication::Get().GetModuleManager().getById<ShadowEngine::RendererModule>(
+            "module:/renderer/vulkan");
```

Comment:
So now you've taken my RendererModule reference to a strict renderer/vulkan reference.. why? This is actively against the point of the RendererModule.

Add a module:/renderer shorthand for the active renderer if you want, but this is actively removing progress.